File: bindings/java-ffm/src/main/java/org/maplibre/nativeffi/StyleImageInfo.java
package org.maplibre.nativeffi;

/** Copied metadata for one runtime style image. */
public record StyleImageInfo(
    int width, int height, int stride, long byteLength, float pixelRatio, boolean sdf) {
  public StyleImageInfo {
    if (width < 0 || height < 0 || stride < 0 || byteLength < 0) {
      throw new IllegalArgumentException("image dimensions and byteLength must be non-negative");
    }
  }
}

File: bindings/kotlin-native/src/nativeMain/kotlin/org/maplibre/nativeffi/style/StyleImageInfo.kt
package org.maplibre.nativeffi.style

/** Copied metadata for one runtime style image. */
public data class StyleImageInfo(
  public val width: Int,
  public val height: Int,
  public val stride: Int,
  public val byteLength: Long,
  public val pixelRatio: Float,
  public val sdf: Boolean,
)
